components/cache/cache_pools.rst

```diff
@@ -0,0 +1,250 @@
+common operations on items, such as saving them or looking for them. Cache pools
+are independent from the actual cache implementation. Therefore, applications
+can keep using the same cache pool even if the underlying cache mechanism
+changes from a filesystem based cache to a Redis or database based cache.
+
+Creating Cache Pools
+--------------------
+
+Cache Pools are created through the **cache adapters**, which are classes that
+implement the :class:`Psr\\Cache\\CacheItemPoolInterface` interface. This
+component provides several adapters ready to use in your applications.
+
+Array Cache Adapter
+~~~~~~~~~~~~~~~~~~~
+
+This adapter is only useful for testing purposes because contents are stored in
+memory and no persisted in any way. Besides, some features explained later are
+not available, such as the deferred saves::
+
+    use Symfony\Component\Cache\Adapter\ArrayAdapter;
+
+    $cache = new ArrayAdapter($defaultLifetime = 0, $storeSerialized = true);
+
+``defaultLifetime``
+    **type**: integer, **default value**: ``0``
+    The default lifetime, in seconds, applied to cache items that don't define
+    their own lifetime. The default value (``0``) means an "infinite" lifetime,
+    but this adapter destroys the cache once the current PHP execution finishes.
+
+``storeSerialized``
+    **type**: boolean, **default value**: ``true``
+    If ``true``, the values saved in the cache are serialized before storing them.
+
+Filesystem Cache Adapter
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This adapter is useful when you want to improve the application performance but
+can't install in the server tools like APC or Redis::
+
+    use Symfony\Component\Cache\Adapter\FilesystemAdapter;
+
+    $cache = new FilesystemAdapter($namespace = '', $defaultLifetime = 0, $directory = null);
+
+``namespace``
+    **type**: string, **default value**: ``''`` (an empty string)
+    The subdirectory created inside the main cache directory (defined in the
+    third argument) to store the cache items.
+
+``defaultLifetime``
+    **type**: integer, **default value**: ``0``
+    The default lifetime, in seconds, applied to cache items that don't define
+    their own lifetime. The default value (``0``) means an "infinite" lifetime,
+    which this adapter respects because items are actually persisted.
+
+``directory``
+    **type**: string, **default value**: ``null``
+    The directory where the cache items are stored as files. Make sure that this
+    directory has read-write permissions for your application. If no directory
+    is defined, a new directory called ``symfony-cache/`` is created in the
+    system's temporary directory.
+
+APCu Cache Adapter
+~~~~~~~~~~~~~~~~~~
+
+This adapter can increase the application performance very significantly, because
+contents are cached in the memory of your server, which is much faster than the
+filesystem. It requires to have installed and enabled the PHP APC extension::
```

Comment:
Note that APCu should be recommended for mostly-append populating scenarios. Having lots of delete&writes creates fragmentation which can degrade performance significantly.